waiter/test/waiter/util/async_utils_test.clj

```diff
@@ -328,3 +328,26 @@
                  (execute task-thread-pool)
                  (async/<!!)))))
     (.shutdown task-thread-pool)))
+
+(deftest test-timer-chan
+  (let [num-iterations 8
+        interval-ms 100
+        tolerance-ms 20
+        call-times-atom (atom [])
+        callback-fn #(swap! call-times-atom conj (System/currentTimeMillis))
+        timer-ch (timer-chan interval-ms)]
+    (async/go
+      (loop []
+        (when (async/<! timer-ch)
+          (callback-fn)
+          (recur))))
+    (Thread/sleep (+ tolerance-ms (* interval-ms num-iterations)))
```

Comment:
tolerance-ms is per chime
do ```(* (+ tolerance-ms interval-ms) num-iterations)```?

Reply:
The tolerance is intended to measure the delay globally.

Reply:
then you use it in ```(every? #(>= % (- interval-ms tolerance-ms)) invocation-diffs)``` which is the tolerance for each tick

Reply:
Any individual call should meet that tolerance as well as globally all calls should also meet that tolerance limit.